todos-app/src/App/App.tsx

```diff
@@ -1,18 +1,49 @@
-import React, { useState } from "react";
+import { useEffect, useState } from "react";
 import NewTodoInput from "./NewTodoInput/NewTodoInput";
 import TodoList from "./TodoList/TodoList";
 import ListFooter from "./ListFooter/ListFooter";
-import { ITodo } from "../DataStructure";
+import { Todo } from "./models/Todos";
+import { FilterType } from "./models/Filters";
 import "../App.css";
 
-const App: React.FC = () => {
-  const [todos, setTodos] = useState<ITodo[]>([]);
+const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterType>("All");
 
-  const addHandler = (input: ITodo) => {
+  useEffect(() => {
+    setFilterTodos(todos);
+    switch (filter) {
+      case "All": {
+        break;
+      }
+      case "Active": {
+        setFilterTodos((prev) =>
+          prev.filter((todo) => todo.completed === false)
+        );
+        break;
+      }
+      case "Completed": {
+        setFilterTodos((prev) =>
+          prev.filter((todo) => todo.completed === true)
+        );
+        break;
+      }
+
+      default:
+        break;
+    }
+  }, [todos, filter]);
```

Comment:
I believe you can simplify this a bit without `useEffect`, displayed todos can be just a constant that will automatically re-calculate when state changes

Reply:
here is an example https://github.com/BimbaMan/TODOs_ReactProject/pull/19

Reply:
Replaced _useEffect_ with recalculating const _filteredTodos_ like in the example